### PR TITLE
E 655 build my contributions list

### PR DESCRIPTION
**Pull request by haydencleary**

### CODE REVIEW

File: frontend/src/components/Contribution/Contribution.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
```
Comment (kaelsky) [inline frontend/src/components/Contribution/Contribution.tsx:1]: Tu n'a plus besoin d'importer `react` depuis React 17 ;)

File: frontend/src/pages/Contributions/index.tsx
Comment (kaelsky) [inline frontend/src/pages/Contributions/index.tsx:1]: `Contributions/Contributions.tsx`

File: frontend/src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import classNames from "classnames";
+
+import { DesktopView, type Variants } from "src/components/Tabs/DesktopView";
+import { MobileView } from "src/components/Tabs/MobileView";
+
+export type Tab = {
+  active: boolean;
+  onClick: () => void;
+  testId?: string;
+} & React.PropsWithChildren;
```
Comment (haydencleary) [inline frontend/src/components/Tabs/Tabs.tsx:10]: Add import

File: frontend/src/App/Layout/Header/View.tsx [hunk 3]
```diff
@@ -67,6 +64,11 @@ export default function HeaderView({
                   {menuItems[RoutePaths.Projects]}
                 </MenuItem>
               )}
+              {menuItems[RoutePaths.Contributions] && (
```
Comment (haydencleary) [inline frontend/src/App/Layout/Header/View.tsx:67]: Use ternary

File: frontend/src/components/ContributionAttribute/ContributionAttribute.tsx [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export function ContributionAttribute({ children }: { children: React.ReactNode }) {
```
Comment (haydencleary) [inline frontend/src/components/ContributionAttribute/ContributionAttribute.tsx:1]: PropsChildren

File: frontend/src/components/Contribution/Contribution.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import { ContributionBadge } from "src/components/ContributionBadge/ContributionBadge";
+import { ContributionIconStatus, ContributionIconStatusType } from "src/components/ContributionIcon/ContributionIcon";
+import { ContributionReview, ContributionReviewStatus } from "src/components/ContributionReview/ContributionReview";
+import { ContributionReward } from "src/components/ContributionReward/ContributionReward";
+import { ContributionBadgeTooltip } from "src/components/ContributionBadgeTooltip/ContributionBadgeTooltip";
+
+type Props = { id: string; title: string; url: string } & React.ComponentProps<typeof ContributionBadge> &
```
Comment (haydencleary) [inline frontend/src/components/Contribution/Contribution.tsx:7]: Import ComponentProps

File: frontend/src/components/ContributionBadgeTooltip/ContributionBadgeTooltip.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+}) {
+  const { T } = useIntl();
+
+  return (
+    <Tooltip anchorId={id} clickable position={TooltipPosition.Top} variant={BackgroundVariant.Blue}>
+      <div className="flex flex-col gap-4 px-1 py-2">
+        {author ? (
+          <div className="flex gap-1 text-xs font-medium text-spaceBlue-200">
+            <span>
+              {T("contributions.tooltip.badgePullRequest")}{" "}
```
Comment (haydencleary) [inline frontend/src/components/ContributionBadgeTooltip/ContributionBadgeTooltip.tsx:36]: &nbsp;

File: frontend/src/components/ContributionTable/ContributionTable.tsx [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+  showHeader?: boolean;
+  status: ContributionTableStatus;
+}) {
+  const { T } = useIntl();
+
+  function renderContribution(contribution: GetAllContributionsQuery["contributions"][number]) {
+    switch (contribution.type) {
+      case ContributionType.Issue:
+        return (
+          <Contribution
```
Comment (haydencleary) [inline frontend/src/components/ContributionTable/ContributionTable.tsx:72]: ternary on `contribution.githubIssue`

File: frontend/src/components/ContributionTable/ContributionTable.tsx [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+      case ContributionType.PullRequest: {
+        let review: ContributionReviewStatus;
+        const codeReviews = contribution?.githubPullRequest?.codeReviews;
+
+        if (codeReviews?.length) {
+          switch (codeReviews[0].outcome) {
+            case null:
+              review = ContributionReviewStatus.UnderReview;
+              break;
+            case "changes_requested":
```
Comment (haydencleary) [inline frontend/src/components/ContributionTable/ContributionTable.tsx:92]: extract to types

File: frontend/.storybook/stories/Contribution.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  },
+  decorators: [
+    withRouter,
+    withMockedProvider(mocks),
+    withAuthProvider({ userId: USER_ID }),
+    withContributorProfilePanelProvider,
+  ],
+};
+
+const mockContribution = {
```
Comment (kaelsky) [inline frontend/.storybook/stories/Contribution.stories.tsx:49]: Use external mocks if you can

File: frontend/src/assets/icons/IssueCancelled.tsx
Comment (kaelsky) [inline frontend/src/assets/icons/IssueCancelled.tsx:1]: Duplicate of `frontend/src/icons/CheckboxCircleLine.tsx`

File: frontend/src/assets/icons/IssueDraft.tsx
Comment (kaelsky) [inline frontend/src/assets/icons/IssueDraft.tsx:1]: If not used we can delete this Icon

File: frontend/src/assets/icons/IssueMerged.tsx
Comment (kaelsky) [inline frontend/src/assets/icons/IssueMerged.tsx:1]: Duplciate of `frontend/src/icons/CheckboxCircleLine.tsx`

File: frontend/src/assets/icons/CodeReviewMerged.tsx
Comment (kaelsky) [inline frontend/src/assets/icons/CodeReviewMerged.tsx:1]: Duplicate of `frontend/src/assets/icons/CodeReviewCheckIcon.tsx`

File: frontend/src/App/Layout/Header/View.tsx (hunk 3 above)
Comment (haydencleary) [reply frontend/src/App/Layout/Header/View.tsx]: 134b6211278d7cbaccd4a30968bbeb698816fb71

File: frontend/src/components/Contribution/Contribution.tsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  GithubPullRequestDraft,
+  GithubPullRequestStatus,
+  QueryContribution,
+} from "src/types";
+
+type Props = {
+  contribution: Pick<
+    QueryContribution,
+    "githubCodeReview" | "githubIssue" | "githubPullRequest" | "id" | "rewardItems" | "type"
+  >;
```
Comment (kaelsky) [inline frontend/src/components/Contribution/Contribution.tsx:21]: Maybe use `ContributionFragment` 
https://github.com/onlydustxyz/marketplace/blob/d7334a7e9447989ba24fd22a4553df0b4352c72f/frontend/src/__generated/graphql.tsx#L18575

File: frontend/src/components/Contribution/Contribution.tsx [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+  const { githubIssue, githubPullRequest, githubCodeReview, id, rewardItems, type } = contribution;
+
+  const title = githubIssue?.title ?? githubPullRequest?.title ?? githubCodeReview?.githubPullRequest?.title ?? "";
+  const htmlUrl =
+    githubIssue?.htmlUrl ?? githubPullRequest?.htmlUrl ?? githubCodeReview?.githubPullRequest?.htmlUrl ?? "";
+  const number = githubIssue?.number ?? githubPullRequest?.number ?? githubCodeReview?.githubPullRequest?.number ?? "";
+  const status = githubPullRequest?.draft
+    ? GithubPullRequestDraft.Draft
+    : ((githubIssue?.status ?? githubPullRequest?.status ?? githubCodeReview?.status ?? "") as GithubItemStatus);
+  const author = (githubIssue?.author ?? githubPullRequest?.author ?? githubCodeReview?.reviewer ?? "") as GithubUser;
```
Comment (kaelsky) [inline frontend/src/components/Contribution/Contribution.tsx:35]: Nice de have, maybe we can factorize this logic using `contribution[type]`

File: frontend/src/components/Contribution/ContributionBadge.tsx [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+  id: string;
+  number: number;
+  type: GithubContributionType;
+  status: GithubItemStatus;
+  title: string;
+  description?: string;
+  author: Pick<GithubUser, "id" | "login" | "avatarUrl">;
+  url: string;
+  withTooltip?: boolean;
+  asAnchor?: boolean;
```
Comment (kaelsky) [inline frontend/src/components/Contribution/ContributionBadge.tsx:34]: `asLink` maybe ?

File: frontend/src/components/Contribution/ContributionReward.tsx [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+  const tooltipId = `${id}-${rewards?.[0].paymentId ?? "rewards"}`;
+
+  return (
+    <>
+      <Tooltip id={tooltipId} clickable position={TooltipPosition.Top} variant={Variant.Blue}>
+        <div className="flex items-center gap-2 px-1 py-2">
+          <Medal2Fill className="flex h-3.5 items-center justify-center text-sm text-orange-400" />
+          <p className="text-sm font-medium leading-none">
+            {T("contributions.tooltip.rewards")}{" "}
+            {rewards.map(({ paymentId }) => `#${paymentId}`.slice(0, 6)).join(", ")}
```
Comment (kaelsky) [inline frontend/src/components/Contribution/ContributionReward.tsx:18]: Externalize this function into an utils ?

File: frontend/src/pages/Contributions/Contributions.tsx [hunk 14]
```diff
@@ -0,0 +1,289 @@
...
+    loading: inProgressLoading,
+    error: inProgressError,
+  } = useGetAllContributionsQuery({
+    variables: {
+      orderBy: sort[GithubContributionStatus.InProgress].orderBy as ContributionsOrderBy,
+      githubUserId,
+      status: GithubContributionStatus.InProgress,
+    },
+    skip: !githubUserId || (!isActiveTab(AllTabs.All) && !isActiveTab(AllTabs.InProgress)),
+    fetchPolicy: "network-only",
```
Comment (kaelsky) [inline frontend/src/pages/Contributions/Contributions.tsx:73]: "cache-first" on nextFetchPolicy ? 

https://www.apollographql.com/docs/react/data/queries/#nextfetchpolicy

**Review by kaelsky — APPROVED** [review]
LGTM, requested changes are nice to have and can be fixed in another PR

File: frontend/src/pages/Contributions/Contributions.tsx (hunk 14 above)
Comment (haydencleary) [reply frontend/src/pages/Contributions/Contributions.tsx]: 55efff1461a1ca313dcfbd4fcd83571360e0d516

File: frontend/src/components/ContributionBadgeTooltip/ContributionBadgeTooltip.tsx (hunk 6 above)
Comment (haydencleary) [reply frontend/src/components/ContributionBadgeTooltip/ContributionBadgeTooltip.tsx]: f0af13b1585b399124cc736c6e60fdea82e04111

File: frontend/src/components/Contribution/Contribution.tsx (hunk 10 above)
Comment (haydencleary) [reply frontend/src/components/Contribution/Contribution.tsx:21]: Can't use the fragment type as it is missing some properties like `rewardItems`.

**Pull request merged**